test-adapter/main.go

```diff
@@ -59,11 +61,19 @@ func main() {
 	cmd.Flags().AddGoFlagSet(flag.CommandLine) // make sure we get the glog flags
 	cmd.Flags().Parse(os.Args)
 
-	provider := cmd.makeProviderOrDie()
-	cmd.WithCustomMetrics(provider)
-	cmd.WithExternalMetrics(provider)
+	testProvider := cmd.makeProviderOrDie()
+	if prov, ok := testProvider.(*fakeprov.TestingProvider); ok {
```

Comment:
instead of casting, have `NewFakeProvider` and `makeProviderOrDie` return both a provider and a webservice object, then just keep the provider implementation private